Gemuzov-SSLSocket

ArtemGemuzov opened the pull request.

p-borozdin commented:
При запуске сервера BrainTalkServerApplication и последующей попытке подключения клиентом BrainTalkClient выдает ошибку 

`javax.net.ssl.SSLException: Connection has closed: javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake`
`Caused by: javax.net.ssl.SSLException: Connection has closed: javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake`
`Caused by: javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake`
`Caused by: java.io.EOFException: SSL peer shut down incorrectly`

При попытке подключиться на сервер через `telnet`выдает ошибку 
`javax.net.ssl.SSLException: Connection has closed: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message`

Skyfor337 committed:
#12 Добавляет maven plugin для сборки и публикации docker образа

  добавили плагин io.fabric8:docker-maven-plugin:0.45.1, так как это наиболее поддерживаемая библиотека на данный момент

avtseben commented on client-cli/src/main/java/BrainTalkClient.java:
```diff
@@ -0,0 +1,47 @@
+
+                in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+                out = new PrintWriter(socket.getOutputStream(), true);
+            }
```
Выходя за скоуп блока try ты закрываешь ресурс помещённый в блок try-with-resources 
то есть закрывается клиентский сокет

По этой причине происходит неожиданное разъединение 
с ошибкой на стороне сервера Remote host terminated the handshake

avtseben commented on server/src/main/java/ru/jafti/braintalk/server/BrainTalkServerApplication.java:
```diff
@@ -2,29 +2,43 @@
+            char[] password = "password".toCharArray();
+
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(getClass().getResourceAsStream("/server.keystore"), password);
```
А где сам файл server.keystore , не подкоммитил?

Commit:
так же поменял порт SSL на 9443 чтобы можно было разделять в будущем открытый сокет и SSL

Commit:
#15 Добавил реализацию клиента

  так же поменял порт SSL на 9443 чтобы можно было разделять в будущем открытый сокет и SSL
